thrust/testing/unique_by_key.cu

```diff
@@ -470,3 +470,58 @@ struct TestUniqueCopyByKeyLargeOutCount
 SimpleUnitTest<TestUniqueCopyByKeyLargeOutCount, IntegralTypes> TestUniqueCopyByKeyLargeOutCountInstance;
 
 #endif // non-OpenMP backend
+
+// Based on GitHub issue: https://github.com/NVIDIA/cccl/issues/1956
+namespace
+{
+struct Entry
+{
+  std::int32_t a;
+  float b;
+};
+using Key = thrust::pair<std::int32_t, Entry>;
```

Comment:
nit: Given that `Key` is a quite common term/keyword in a TU concerned with `unique_by_key` and this `Key` is really just the `Key` in one specific test case, should we move this type alias inside `TestKeysWithoutEqualityOperator` and just make `Key` a template parameter on `Compare::operator()` to avoid potential (future) conflicts?

Reply:
Sounds fair. Applied.